### PR TITLE
[MAINTENANCE] Add pre-commit autofix to PRs

### PR DESCRIPTION
**Pull request by cdkini**
Have pre-commit automatically update PRs based on linting violations
Let pre-commit fix pre-existing violations

- [x] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [x] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [x] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [x] Appropriate tests and docs have been updated

For more information about contributing, see [Contribute](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

**Review by Kilo59 — APPROVED** [review]

**Comment by ghost** [comment]
<details open>
<summary><b>👇 <a href="https://app.codesee.io/r/reviews?pr=8873&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations">Click on the image</a> for a new way to code review</b>
</summary>

####

[Review these changes using an interactive CodeSee Map](https://app.codesee.io/r/reviews?pr=8873&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations)

#### Legend
CodeSee Map legend
</details>

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -4,6 +4,9 @@ repos:
     hooks:
       - id: check-ast
       - id: check-json
+        exclude: invalid_json_file.json |
+              tests/test_sets/fixed_distributional_test_dataset.json |
+              tests/test_sets/expected_cli_results_default.json
```
Comment (Kilo59) [inline .pre-commit-config.yaml]: The first file in this list is intentionally invalid json but it's less obvious that the other 2 were meant to be.

**Comment by Kilo59** [comment]
@cla-bot check

File: tests/render/renderer/v3/test_suite_edit_notebook_renderer.py
Comment (Kilo59) [inline tests/render/renderer/v3/test_suite_edit_notebook_renderer.py:1]: I don't understand why I had to make these changes 🤷

**Pull request merged**